pkg/plugin/system/system_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright © 2022 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package system
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	buildeventstream "aspect.build/cli/bazel/buildeventstream/proto"
+	rootFlags "aspect.build/cli/pkg/aspect/root/flags"
+	"aspect.build/cli/pkg/ioutils"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+)
+
+// A plugin to simply log plugin callbacks
+type LoggerPlugin struct {
```

Comment:
I think what you want is a mock of the `Plugin` interface. Take a look at for example https://github.com/aspect-build/aspect-cli/blob/2cc2cb276b936077943fd30f3f6aa6fa46acb069/pkg/aspect/build/build_test.go.

You can generate mocks like https://github.com/aspect-build/aspect-cli/tree/2cc2cb276b936077943fd30f3f6aa6fa46acb069/pkg/bazel/mock.

Reply:
This will change substantially the rest of your tests, but I think it will be cleaner and more standardized.

Reply:
Yeah this thing is a bit of a mess, and when expanding the tests more (testing error handling being a big one) it would just get worse. Hopefully mocks make that better, I'll take a look 👍